typescript/packages/subsurface-viewer/src/components/Map.tsx

```diff
@@ -647,6 +654,9 @@ const Map: React.FC<MapProps> = ({
     }, [triggerHome]);
 
     useEffect(() => {
+        if (didUserChangeCamera && !isCameraPositionDefined) {
+            return;
+        }
         const [Views, viewStates] = createViewsAndViewStates(
```

Comment:
Is what you want to achieve that `createViewsAndViewStates` is only called on first render and if camera is defined?

Reply:
The aim is to not call "createViewsAndViewStates  (and hence reset the camera) if the user has manipulated the camera by interacting with it.

Reply:
I think `onViewStateChanged` could be called for other reasons, like eg. syncing cameras across components, so maybe `didUserChangeCamera` should be named something like `viewStateChanged`?